connection.go

```diff
@@ -142,18 +142,29 @@ func (c *Connection) Connect() error {
 		return nil
 	}
 
-	d := &net.Dialer{Timeout: c.Opts.ConnectTimeout}
+	d := &net.Dialer{
+		Timeout: c.Opts.ConnectTimeout,
+		// The dialer KeepAlive option will set both the time the connection is idle
+		// before probes are sent and the interval between the probes.
+		KeepAlive: c.Opts.KeepAlive,
+	}
 
 	if c.Opts.TLSConfig != nil {
 		conn, err = tls.DialWithDialer(d, "tcp", c.addr, c.Opts.TLSConfig)
 	} else {
 		conn, err = d.Dial("tcp", c.addr)
 	}
-
 	if err != nil {
 		return fmt.Errorf("connecting to server %s: %w", c.addr, err)
 	}
 
+	if tcpConn, ok := conn.(*net.TCPConn); ok {
+		err = tcpConn.SetNoDelay(false)
+	}
+	if err != nil {
+		return fmt.Errorf("setting TCP_NODELAY: %w", err)
+	}
```

Comment:
Only check err if we modified it. 
```suggestion
		err = tcpConn.SetNoDelay(false)
		if err != nil {
			return fmt.Errorf("setting TCP_NODELAY: %w", err)
		}
```